client/consensus/grandpa/changeTree.go

```diff
@@ -0,0 +1,453 @@
+// Copyright 2023 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package grandpa
+
+import (
+	"errors"
+	"fmt"
+	"github.com/ChainSafe/gossamer/pkg/scale"
+	"golang.org/x/exp/constraints"
+)
+
+/*
+	TODO give a summary of how this works in context of grandpa
```

Comment:
can we add this description before merging

Reply:
I have given a small description, let me know if more is desired